portfolio/src/main/java/com/google/sps/data/ArrayListTimeRangeGroup.java

```diff
@@ -0,0 +1,171 @@
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+/** Models an implementation of the TimeRangeGroup model using ArrayList. */
+public class ArrayListTimeRangeGroup implements TimeRangeGroup {
+
+  // This list of all the time ranges will be sorted by start time ascending
+  // in add and delete methods.
```

Comment:
Consider clarifying in the comment that the time ranges in allTimeRanges are always pair-wise disjoint.

Reply:
Updated.